Extensions/Effects/JsExtension.js

```diff
@@ -274,6 +273,39 @@ module.exports = {
         .setType('number')
     );
 
+    const bulgePinchEffect = extension
+      .addEffect('BulgePinch')
+      .setFullName(_('Bulge Pinch'))
+      .setDescription(_('Bulges or pinches the image in a circle.'))
+      .addIncludeFile('Extensions/Effects/pixi-filters/filter-bulge-pinch.js')
+      .addIncludeFile('Extensions/Effects/bulge-pinch-pixi-filter.js');
+    const bulgePinchProperties = bulgePinchEffect.getProperties();
+    bulgePinchProperties.set(
+      'centerX',
+      new gd.PropertyDescriptor(/* defaultValue= */ '0.5')
+        .setLabel(_('Center X (between 0 and 1, 0.5 is window middle)'))
```

Comment:
I know this was done in other filters - but I think we should use center X/Y in pixels, rather than a percentage of the window size.

The reason is that filters will soon work on objects - so there is no "window" size in this context.